docs/blog/_posts/2020-12-18-scala3-m3.md

```diff
@@ -0,0 +1,203 @@
+  ...
+given listOrd[T: Ordering] as Ordering[List[T]]:
+  ...
+
+given Ordering[Int]:
+  ...
+given [T: Ordering] as Ordering[List[T]]:
+  ...
+
+given global as ExecutionContext = ForkJoinContext()
+given Context = ctx
+```
+
+Here is how the above is expressed now:
+
+```scala
+given intOrd: Ordering[Int] with
+  ...
+given listOrd[T: Ordering]: Ordering[List[T]] with
+  ...
+
+given Ordering[Int] with
+  ...
+given [T: Ordering]: Ordering[List[T]] with
+  ...
+
+given global: ExecutionContext = ForkJoinContext()
+given Context = ctx
+```
+
+You can find an extensive discussion on the above change in its [PR #10538](https://github.com/lampepfl/dotty/pull/10538).
+
+## Drop `as` in patterns
+Since we dropped `as` from `given`s, we lost a strong reason for having `as` at all. For this reason, we dropped them also from patterns. If previously the correct binding syntax is patterns was:
+
+```scala
+case opt as Some(foo)
+```
+
+Now it got reverted to the Scala 2 style:
+
+```scala
+case opt @ Some(foo)
+```
+
+## Switch back to the old context function closure syntax
+Previously, context function closures were written as follows:
+
+```scala
+(using s: Show[String]) => s.show("foobar")
+```
+
+Now, we have reverted to the following syntax:
+```scala
+(s: Show[String]) ?=> s.show("foobar")
+```
+
+# `Matchable` trait – a new top-level type
+The motivation for this change is the fact that it is possible to access the underlying type referenced by an opaque type using pattern matching:
+
+```scala
+val imm = IArray(1,2,3) // supposedly immutable...
+imm match
+   case a: Array[Int] => a(0) = 0  // but that's shown to be lie
+```
+
+To address this change, a new trait, `Matchable`, is introduced at the top of the type hierarchy:
+
+```scala
+abstract class Any:
+  def asInstanceOf
+  def ==
+  def !=
+  def ##
+  def equals
+  def hashCode
+  def toString
+
+trait Matchable extends Any:
+  def isInstanceOf
+  def getClass
+
+class AnyVal extends Any, Matchable
+
+class Object extends Any, Matchable
+```
+
+With the addition of this type, we now make the compiler emit warnings when trying to do a trick like the one discussed above:
+
+```scala
+-- Warning: i7314.scala:6:12 ---------------------------------------------------
+6 |    case a: Array[Int] =>
+  |            ^^^^^^^^^^
+  |            pattern selector should be an instance of Matchable,
+  |            but it has unmatchable type opaques.IArray[Int] instead
+```
+
+The change involved an extensive discussion which you can read in the [PR #10670](https://github.com/lampepfl/dotty/pull/10670).
+
+# Tooling improvements
+As the work on Scala 3 gets closer to the stable release, the focus increasingly shifts on the tooling available to get started with Scala 3.
+
+As part of the tooling effort, Scala 3 documentation tool is rapidly improved. [PR #10522](https://github.com/lampepfl/dotty/pull/10522) proves that the doctool can generate documentation for the community build projects.
```

Comment:
I think we should mention that we moved away from dottydoc and we are using scala3doc by default now.

Reply:
Seconding that. Judging by community build projects disabling the `doc` task, ppl had bad experience with Dottydoc previously, so it'd be nice to mention that the default has changed and that they can give generating documentation another shot.